### ISSUE TITLE
Local GeoJSON or MVT as data provider

### ISSUE THREAD

**Issue by borsje** [tangrams/tangram#636]
My goal is to create a fully offline LeafletJS map. I have my data available locally, but not in rendered .pngs. Loading GeoJSON in LeafletJS is no option, as styling is very limited, the map quickly becomes slow, etc.
Using Tangram seems to be best solution, but I can't get it to load my local data. Using a Nextzen online source provides no problems.

**TANGRAM VERSION:** 0.15.1 (currently latest)

**ENVIRONMENT:** Chromium (Electron)

**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

Recreate the basic example and add an additional layer to scene.yaml as follows:

```layers:
    ...
    dynamic:
        data: { source: dynamic }
        draw:
            lines:
                order: 100
                color: '#cc6666'
                width: 8
```

Dynamically add JSON data to map somewhere in the code (mine is added when I move the LeafletJS map (and the view reaches a part of the map that hasn't been loaded yet)):

```function addMap(latitude_index, longitude_index) {
    ...
    layer.scene.setDataSource("dynamic", {type: "GeoJSON", data: json});
}
```

`json` has the following format (I am showing the JSON equivalent, but it's a JS object):

```{
    "type": "FeatureCollection",
    "features": [{
        "type": "Feature",
        "properties": { ... }
        "geometry": {
            "type": "LineString",
            "coordinates": [[6.2491176, 49.5443152], ...]
        }
    }, ...]
}
```

**RESULT:**

No data is added; the map is completely empty when I also remove my default Nextzen source. There are no errors.

**EXPECTED RESULT:**

I want my scene.yaml without any sources and completely build the map from local JSON or MVT (I have it stored in my own format, so converting has to be done anyway).
Storing my local data as slippy map tiles seems to be too extreme, but theoretically could be possible. I would rather dynamically load my data.

**Comment by bcamper** [comment]
Your example looks correct on first glance. Would you be willing to provide
complete sample scene and data files that don't load as expected?

**Comment by borsje** [comment]
I continued my debugging in the meanwhile and got very lucky: as I randomly hit the zoom out button I saw the rendered roads for a fraction of a second. I realised I basically loaded/rendered an out of view part of my data using .setDataSource. I was lucky I saw it as it was exactly located on the "border" where it got hidden due to the low zoom level.

In addition I needed to provide my GeoJSON layered to work with the default scene.yaml:

```
{
    roads: {
        type: "FeatureCollection",
        features: [ ... ]
    }
}
```

So the bugs ended up being purely due to my own code :), thanks for the quick reply!
